collated/test/thehelpfulbees.md

```diff
@@ -0,0 +1,149 @@
+# thehelpfulbees
+###### \java\seedu\address\logic\commands\FindCommandTest.java
+``` java
+
+    @Test
+    public void execute_searchByTag_noPersonFound() throws Exception {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 0);
+        FindCommand command = prepareCommand("t\\worstEnemy");
+        assertCommandSuccess(command, expectedMessage, Collections.emptyList());
+    }
+
+    @Test
+    public void execute_searchByTag_onePersonFound() throws Exception {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 1);
+        FindCommand command = prepareCommand("t\\owesMoney");
+        assertCommandSuccess(command, expectedMessage, Arrays.asList(BENSON));
+    }
+
+    @Test
+    public void execute_searchByTag_multiplePeopleFound() throws Exception {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 7);
+        FindCommand command = prepareCommand("t\\friends");
+        assertCommandSuccess(command, expectedMessage,
+                Arrays.asList(ALICE, BENSON, CARL, DANIEL, ELLE, FIONA, GEORGE));
+    }
+
+    /**
+     * Parses {@code userInput} into a {@code FindCommand}.
+     */
+    private FindCommand prepareCommand(String userInput) {
+
+        //this is work that the parser will do
+        String[] nameKeywords = userInput.split("\\s+");
+        Predicate<ReadOnlyPerson> searchPredicate;
+        if (nameKeywords.length == 1 && nameKeywords[0].startsWith("t\\")) {
+            searchPredicate  = new TagsContainKeywordPredicate(nameKeywords[0].substring(2));
+        } else {
+            searchPredicate = new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords));
+        }
+
+        FindCommand command = new FindCommand(searchPredicate);
+        command.setData(model, new CommandHistory(), new UndoRedoStack());
+        return command;
+    }
+
+    //@@
+
+    /**
+     * Asserts that {@code command} is successfully executed, and<br>
+     *     - the command feedback is equal to {@code expectedMessage}<br>
+     *     - the {@code FilteredList<ReadOnlyPerson>} is equal to {@code expectedList}<br>
+     *     - the {@code AddressBook} in model remains the same after executing the {@code command}
+     */
+    private void assertCommandSuccess(FindCommand command, String expectedMessage, List<ReadOnlyPerson> expectedList)
+            throws Exception {
+        AddressBook expectedAddressBook = new AddressBook(model.getAddressBook());
+        CommandResult commandResult = command.execute();
+
+        assertEquals(expectedMessage, commandResult.feedbackToUser);
+        assertEquals(expectedList, model.getFilteredPersonList());
+        assertEquals(expectedAddressBook, model.getAddressBook());
```

Comment:
These 3 lines seem to be used in lots of tests. Considering putting them in util methods.